tests/e2e/volumeproperty_test.go

```diff
@@ -0,0 +1,175 @@
+	s.dockerHostName = govc.RetrieveVMNameFromIP(s.dockerHostIP)
+	s.esxIP = inputparams.GetEsxIP()
+	s.formatTypes = []string{"thin", "zeroedthick", "eagerzeroedthick"}
+	s.dockerVolumeRemoveCmd = dockerconst.RemoveVolume
+	s.vmAttached = false
+	s.verifyParams(c)
+
+}
+
+// verifyParams - skips the test if dockerHostIP/dockerHostName/esxIP are empty
+func (s *VolumePropVerificationSuite) verifyParams(c *C) {
+	if s.dockerHostIP == "" || s.dockerHostName == "" || s.esxIP == "" {
+		c.Skip("Empty values - hence skipping the test")
+	}
+}
+
+// TearDownTest - Removes all the containers and deletes the volumes
+func (s *VolumePropVerificationSuite) TearDownTest(c *C) {
+	out, err := dockercli.RemoveAllContainers(s.dockerHostIP)
+	c.Assert(err, IsNil, Commentf(out))
+	out, err = ssh.InvokeCommand(s.dockerHostIP, s.dockerVolumeRemoveCmd)
+	c.Assert(err, IsNil, Commentf(out))
+}
+
+var _ = Suite(&VolumePropVerificationSuite{})
+
+/*
+Test steps:
+1. SSH to a vm and create volumes of size 100 mb and specified disk format ["thin", "zeroedthick", "eagerzeroedthick"]
+2. Do docker inspect on the volume get size, disk format and attached-to-vm field.
+   Expected value: {"100MB", "thin/zeroedthick/eagerzeroedthick", no-value} respectively.
+3. SSH to the esx and get capacity, disk format and attached-to-vm field for the volume using Admin cli
+   Expected value: {"100MB", "thin/zeroedthick/eagerzeroedthick", "detached"} respectively.
+4. Verifies values obtained from admin cli and docker cli.
+5. SSH to the vm and run a container and mount the volume created.
+6. SSH to the vm and esx and verify the attached-to-vm field for volume - both docker cli and admin cli values should be same.
+7. Again verify capacity and disk format using docker cli and admin cli to make sure things are fine after running the container.
+
+NOTE: Do steps 5 ,6 and 7 only for volume of 'thin' disk format
+*/
+
+func (s *VolumePropVerificationSuite) TestVolumeProperties(c *C) {
+	log.Printf("START: TestVolumeProperties")
+
+	// creating volumes of all disk formats
+	s.createVolumes(c)
+	for _, volumeName := range s.volumeNames {
+		// getting and verifying the admin cli and docker cli values
+		s.getPropertiesAndVerify(c, volumeName)
+		s.dockerVolumeRemoveCmd = s.dockerVolumeRemoveCmd + " " + volumeName
+	}
+
+	// attaching only thin volume and verifying its properties
+	out, err := dockercli.AttachVolume(s.dockerHostIP, s.volumeNames[0], s.containerName)
+	c.Assert(err, IsNil, Commentf("Failed to attach the volume [%s]", out))
+	s.vmAttached = true
+	s.getPropertiesAndVerify(c, s.volumeNames[0])
+
+	log.Printf("END: TestVolumeProperties")
+}
+
+// createVolumes - creates volumes of each format type
+func (s *VolumePropVerificationSuite) createVolumes(c *C) {
+	for i, formatType := range s.formatTypes {
+		s.containerName = inputparams.GetContainerNameWithTimeStamp(TestName)
+		s.volumeNames[i] = inputparams.GetVolumeNameWithTimeStamp(TestName)
```

Comment:
Watchout, the GetVolumeNameWithTimeStamp() is using time.New().Unix() which returns the number of seconds since some time. Successive calls can return the same name. I'd seen this and use names with a random generated number attached to it. For clarity sake you could have different names - thick, zeroedthick, thin  and attach the timestamp.